`AmpModal` implementation

egormesyats opened the pull request:
New `AmpModal` component which uses `<amp-lightbox>` inside

Related: 
https://moovweb.atlassian.net/browse/ANE-733 
https://github.com/moovweb-projects/annieselke/pull/523

markbrocato commented on packages/react-storefront/src/amp/AmpModal.js:
```diff
@@ -0,0 +1,85 @@
+  }
+
+  static defaultProps = {
+    id: 'moov-amp-lightbox',
```
Is there a reason why this component needs an id?  Generally I like to stay away from ids to prevent collisions with other components that may have the same id.

egormesyats replied:
Yes, it definitely needs it. It's a **required** attribute:
https://www.ampproject.org/docs/reference/components/amp-lightbox#attributes

markbrocato replied:
Let's mark `id` as a required prop, but not provide a default then.  The developer is going to need to choose an id anyway so they can open it with an amp bind event.  Having a default id is a bit surprising and doesn't add any value other than delaying the developer's realization that he needs to assign an id to properly use the component.

egormesyats replied:
Yea. It's even better!

markbrocato commented on packages/react-storefront/src/amp/AmpModal.js:
```diff
@@ -0,0 +1,85 @@
+  static defaultProps = {
+    id: 'moov-amp-lightbox',
+    animateIn: 'fade-in',
+    ampProps: {},
```
There is no need to have a separate ampProps object if we are just going to spread all other props onto the amp-lightbox component.  Just add "All props not specifically documented here are spread to the amp-lightbox element." to the class comments.

egormesyats replied:
Makes sense 👍. I'll fix this

markbrocato commented:
Please add a snapshot test in jest for this component.

egormesyats commented:
- Removed `ampProps`
- Jest tests added
---
I also removed `zIndex` from `AmpModel` because we don't really need it and tests are failing for some reason:

`TypeError: Cannot read property 'modal' of undefined

      10 | export const styles = theme => ({
      11 |   root: {
    > 12 |     zIndex: theme.zIndex.amp.modal + 2, // zIndex of AppBar's withAmp + 1`
